packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/Legend/ReferenceLine.tsx

```diff
@@ -188,6 +189,72 @@ const ReferenceLineValue: FC<ReferenceLineValueProps> = ({
     );
 };
 
+type AccordionControlProps = {
+    label: string;
+    lineColor: string;
+    onControlClick: () => void;
+    onDelete: () => void;
+    onColorChange: (c: string) => void;
+} & MantineAccordionControlProps;
+
+const AccordionControl: FC<AccordionControlProps> = ({
+    label,
+    lineColor,
+    onControlClick,
+    onDelete,
+    onColorChange,
+    ...props
+}) => {
+    const { ref, hovered } = useHover<HTMLDivElement>();
+    const { colorPalette } = useVisualizationContext();
+    return (
+        <Group
+            noWrap
+            ref={ref}
+            spacing="one"
```

Comment:
more than happy to go further and refactor this into an shared component (between `ReferenceLine` and `ConditionalFormatting`. LMK what you think @magnew !

Reply:
I definitely think we should do that. This same problem exists there and this is a good answer. If you don't have time, maybe I can pick it up.

Reply:
And here it is @magnew ! https://github.com/lightdash/lightdash/pull/9679/commits/5e657daba5c1c76518ea5944453048c058e2b924